flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/EncodingUtils.java

```diff
@@ -187,6 +187,33 @@ public static String hex(byte[] bytes) {
         return new String(hexChars);
     }
 
+    /**
+     * Inspired from {@link EncodingUtils#decodeHex(String)}, but returns null instead of throwing
```

Comment:
It would be good to emphasize specific processing differences, including header byte strategies for odd-length inputs.
Another thing is the method name, I'm thinking about using a consistent name, e.g., `decodeHexLax` or `lenientHexDecode `, WDYT?

Reply:
Perhaps we can phrase like this:
```
 * Converts an array of characters representing hexadecimal values into an array of bytes of
 * those same values. The returned array will be half the length of the passed array, as it
 * takes two characters to represent any given byte. If the input array has an odd length,
 * the first byte is handled separately and set to 0.
 *
 * <p>Unlike {@link #decodeHex(String)}, this method does not throw an exception for odd-length
 * inputs or invalid characters. Instead, it returns null if invalid characters are encountered.
 *
 * @param bytes An array of characters containing hexadecimal digits.
 * @return A byte array containing the binary data decoded from the supplied char array,
 *         or null if the input contains invalid hexadecimal characters.
```

Reply:
Should we keep consistent with existed method `hex`, as with the built-in functions `HEX` and `UNHEX`?
And maybe we can create a function `unhex(byte[] bytes, boolean throwOnFailure)` to call both implementations and change the signature of `decodeHex` to `byte[]`, because in the original implementation `String` is also converted to a char array for calling. This also ensures performance for `BinaryStringData`.